src/Plugin/Drupal7/Audit/Database/DatabaseConnectionAnalysis.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Drutiny\Plugin\Drupal7\Audit;
+
+use Drutiny\Audit\AbstractAnalysis;
+use Drutiny\Sandbox\Sandbox;
+
+/**
+ * Provides all database connection information.
+ */
+class DatabaseConnectionAnalysis extends AbstractAnalysis {
+
+  public function gather(Sandbox $sandbox) {
+    $db_connections = $this->target->getService('drush')
+      ->runtime(function() {
+        global $databases;
+        return $databases;
+    });
+    $this->unset_user_pass($db_connections, ['username', 'password']);
+    $this->set('database_connections', $db_connections);
+  }
+
+  /**
+   * Helper function to unset non-required keys from connections array.
+   * @param array $connections
+   * @param array $unwanted_keys
+   */
+  public function unset_user_pass(array &$connections, array $unwanted_keys) {
```

Comment:
Unsetting credentials here means the credential data has already been transfered back to Drutiny. So while it won't be available at the templating layer, it has been aggregated and stored in the caching layer. See comment above as to why we should remove them in the runtime environment instead.

Reply:
Done.